api/src/main/java/com/velocitypowered/api/proxy/config/ProxyConfig.java

```diff
@@ -0,0 +1,41 @@
+package com.velocitypowered.api.proxy.config;
+
+import com.velocitypowered.api.util.Favicon;
+import net.kyori.text.Component;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public interface ProxyConfig {
```

Comment:
Make sure to add in Javadoc.

Reply:
I went ahead and wrote javadoc for a methods, not the best so please feel free to make them not suck so much